src/lua/httpc.lua

```diff
@@ -529,6 +529,8 @@ end
 --
 --      params - a table with query parameters;
 --
+--      http_version - describes a http version;
```

Comment:
This option is not tested at all.

I propose to check at least a version that http client is enforce in request:
Something like this:

1st console:
```
[0] ~$ netcat -l 127.0.0.1 8080
HEAD / HTTP/1.1
Host: 127.0.0.1:8080
User-Agent: curl/7.81.0
Accept: */*
...
```

2nd console:
```
[0] ~$ curl -I 127.0.0.1:8080
```

This will not test support of different protocol versions, but will test that protocol version enforced by client is correct.

We can implement TCP echo server using builtin `socket` module in Tarantool, https://www.tarantool.io/en/doc/latest/reference/reference_lua/socket/#socket-tcp-server

Reply:
Nice idea.

After your message I've thought on it a bit more and remembered that there is the `verbose = true` option, which writes request details on stderr. We can run tarantool using the `test.justrun` helper with a script that runs an HTTP request. Then, analyze the stderr output in a test case.

Maybe this variant doesn't require an alive listening socket at all. At least, it worth to try.

Reply:
I've experimented a bit with it: we still need a TCP server, but it is enough to have a simple one, which just accepts and closes a connection.

It looks like we can differentiate all the needed situations using the verbose output:

* `http://`: HTTP 1.1 vs HTTP/1.1 + Upgrade to HTTP/2 vs HTTP/2
* `https://`: HTTP/1.1 vs HTTP/2

Reply:
Ilya added the tests. Sergey, do you want to glance on them?

Reply:
added comments regarding tests